### ISSUE TITLE
Get a proper CI with all functions

### ISSUE THREAD

**Issue by monperrus** [eclipse/repairnator#1152]
* [x] activate Eclipse CI
* [x] remove skipTests in pipeline CI
* [x] remove skipTests in maven-repair  CI

**Comment by monperrus** [comment]
Travis CI seems dead, now blocking progress.

@henry-lp you did some experiments on https://ci.eclipse.org/repairnator/, what's the status?
@javierron do you have access to https://ci.eclipse.org/repairnator/ now that you're committer?

**Comment by javierron** [comment]
@monperrus Yes, I can log in and the features seem to be available to me.

**Comment by monperrus** [comment]
nice, do you want to give and try and write a Jenkins CI job for Repairnator based on the Travis config?

**Comment by henry-lp** [comment]
> @henry-lp you did some experiments on https://ci.eclipse.org/repairnator/, what's the status?

@monperrus, we tried to install Repairnator as a Jenkins plugin before but Eclipse did not have any intention of allowing an installation of third-party plugin. I think we would need to ask them everytime to do that, so I think that idea was putting on hold after that.

I would be glad to help in the normal case but I unfortunately don't have lots of spare time due to the need of preparing the thesis presentation and work.

**Comment by javierron** [comment]
@monperrus @henry-lp I can set it up. Have to learn my way around Jenkins first but should not take more than a day or two.

**Comment by monperrus** [comment]
Super! Thanks Javier.

**Comment by javierron** [comment]
The current setup of Jenkins is still far from complete. We still need to:
- Enable testing
- Running parallel jobs is causing failures:
  - Simultaneous branches (e.g. master + pull request)
  - Simultaneous stages in build

**Comment by monperrus** [comment]
Fully agree. The highest priority is to re-enable the pipeline tests. @javierron @andre15silva who feels like looking at this?

**Comment by andre15silva** [comment]
I can take a look at it this weekend.

**Comment by andre15silva** [comment]
Both tests in https://github.com/eclipse/repairnator/blob/master/src/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/pipeline/TestPipelineSequencerRepair.java are failing.

`TestPipelineBuildPassBranch` has `PATCH_NOT_FOUND`, while `TestPipelineBuildFailBranch` has `NOTBUILDABLE`.

Another weird thing is that it creates two SequencerRepair steps.

@javierron do you know what might be causing this?

Example:

`java -cp $TOOLS_JAR:target/repairnator-pipeline-3.3-SNAPSHOT-jar-with-dependencies.jar fr.inria.spirals.repairnator.pipeline.Launcher --build 703887431 --sequencerRepair`

produces

```
[main] 17:08:16.760 INFO  f.i.s.r.p.i.ProjectInspector - ----------------------------------------------------------------------
[main] 17:08:16.760 INFO  f.i.s.r.p.i.ProjectInspector - PIPELINE EXECUTION SUMMARY
[main] 17:08:16.760 INFO  f.i.s.r.p.i.ProjectInspector - ----------------------------------------------------------------------
[main] 17:08:16.760 INFO  f.i.s.r.p.i.ProjectInspector - CloneRepository.......................................SUCCESS [  2 s ]
[main] 17:08:16.760 INFO  f.i.s.r.p.i.ProjectInspector - CheckoutBuggyBuild....................................SUCCESS [  0 s ]
[main] 17:08:16.760 INFO  f.i.s.r.p.i.ProjectInspector - BuildProject..........................................SUCCESS [ 16 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - InitRepoToPush........................................SKIPPED
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - SequencerRepair...............................PATCH_NOT_FOUND [  2 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - TestProject...........................................SUCCESS [  3 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - GatherTestInformation.................................SUCCESS [  0 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - ComputeClasspath......................................SUCCESS [ 12 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - ComputeSourceDir......................................SUCCESS [  0 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - ComputeTestDir........................................SUCCESS [  0 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - SequencerRepair.......................................NOT RUN
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - BranchingStep.........................................SUCCESS [  0 s ]
[main] 17:08:16.761 INFO  f.i.s.r.p.i.ProjectInspector - CommitPatch...........................................SKIPPED
[main] 17:08:16.762 INFO  f.i.s.r.p.i.ProjectInspector - ComputeModules........................................SUCCESS [  0 s ]
[main] 17:08:16.762 INFO  f.i.s.r.p.i.ProjectInspector - WritePropertyFile.....................................SUCCESS [  0 s ]
[main] 17:08:16.762 INFO  f.i.s.r.p.i.ProjectInspector - CommitProcessEnd......................................SKIPPED
[main] 17:08:16.762 INFO  f.i.s.r.p.i.ProjectInspector - PushProcessEnd........................................SKIPPED
[main] 17:08:16.762 INFO  f.i.s.r.p.i.ProjectInspector - ----------------------------------------------------------------------
[main] 17:08:16.762 INFO  f.i.s.r.p.i.ProjectInspector - PIPELINE FINDING: TEST FAILURE
[main] 17:08:16.762 INFO  f.i.s.r.p.i.ProjectInspector - ----------------------------------------------------------------------
```

**Comment by javierron** [comment]
Hey @andre15silva, thanks for checking this! 

> TestPipelineBuildPassBranch has PATCH_NOT_FOUND, while TestPipelineBuildFailBranch has NOTBUILDABLE.

Those tests do not check the status of the pipeline, so it is not very important here. Can you tell me which assertions are the ones failing?

> Another weird thing is that it creates two SequencerRepair steps.

This is intentional. The SequencerRepair pipeline has two branches, each branch has its own SequencerRepair step. You can check how it's assembled [here](https://github.com/eclipse/repairnator/blob/master/src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/components/RunInspector4SequencerRepair.java). The list view of the steps is not yet compatible with branched pipelines.

**Comment by javierron** [comment]
Related: Possibly we are going to have some tests failing when travis-ci.org stops working next month.

I can think of the ones that read directly from travis builds. e.g.:
 
`Launcher launcher = new Launcher(new String[]{"--build", "703887431", "--sequencerRepair" });` on [this](https://github.com/eclipse/repairnator/blob/master/src/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/pipeline/TestPipelineSequencerRepair.java) test.

**Comment by andre15silva** [comment]
> Those tests do not check the status of the pipeline, so it is not very important here. Can you tell me which assertions are the ones failing?

It's the last assertion, `assertNull`. I don't think that's correct, but the way I understood it, even the correct version wouldn't work.

My idea is that is should be something like this, on the first one:
```
        AbstractStep step =  launcher.getInspector().getSteps().get(10); //test fix sequencer repair
        assertNotNull(step);
        assertNotNull(step.getStepStatus());
        assertEquals(step.getStepStatus().getStatus(), StepStatus.StatusKind.SUCCESS);
```

**Comment by andre15silva** [comment]
> Related: We are possibly going to have some tests failing when travis-ci.org stops working next month.
> 
> I can think of the ones that read directly from travis builds. e.g.:
> 
> `Launcher launcher = new Launcher(new String[]{"--build", "703887431", "--sequencerRepair" });` on [this](https://github.com/eclipse/repairnator/blob/master/src/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/pipeline/TestPipelineSequencerRepair.java?rgh-link-date=2020-12-14T19%3A04%3A42Z) test.

Yup, that is going to be a problem. We should try to migrate the tests before that happens.

**Comment by javierron** [comment]
> My idea is that is should be something like this, on the first one

That makes sense. Although the last assertion is not necessary, because in this test we don't check that the tool succeeds, instead we check that the pipeline takes the correct branch.

**Comment by andre15silva** [comment]
> That makes sense. Although the last assertion is not necessary, because in this test we don't check that the tool succeeds, instead we check that the pipeline takes the correct branch.

Oh okay, I thought this was supposed to test if the repair worked as well. But anyway, it was supposed to work right?

**Comment by javierron** [comment]
> But anyway, it was supposed to work right?

Yes.

Perhaps the problem has something to do with docker (which SequencerRepair needs), because I think it is not available right now in our CI environment.

**Comment by andre15silva** [comment]
> Yes.
> 
> Perhaps the problem has something to do with docker (which SequencerRepair needs), because I think it is not available right now in our CI environment.

I got the same problem running locally `java -cp $TOOLS_JAR:target/repairnator-pipeline-3.3-SNAPSHOT-jar-with-dependencies.jar fr.inria.spirals.repairnator.pipeline.Launcher --build 703887431 --sequencerRepair` tho. Could you check to see if you can reproduce it?

**Comment by monperrus** [comment]
Closed per #1166
